### PR TITLE
[YANG]: Update port Yang models to support multi-asic platform

### LINKED ISSUE #9580: [yang-models] port validation issue
# name          lanes                     alias       index speed     asic_port_name role
Ethernet13      1280,1281,1282,1283       Eth1/1/13   13    100000    Eth13-ASIC1  Ext
Ethernet14      1284,1285,1286,1287       Eth1/1/14   14    100000    Eth14-ASIC1  Ext
Ethernet16      1288,1289,1290,1291       Eth1/1/16   16    100000    Eth16-ASIC1  Ext
Ethernet17      1292,1293,1294,1295       Eth1/1/17   17    100000    Eth17-ASIC1  Ext
Ethernet19      1024,1025,1026,1027       Eth1/1/19   19    100000    Eth19-ASIC1  Ext
Ethernet20      1036,1037,1038,1039       Eth1/1/20   20    100000    Eth20-ASIC1  Ext
Ethernet22      1028,1029,1030,1031       Eth1/1/22   22    100000    Eth22-ASIC1  Ext
Ethernet23      1032,1033,1034,1035       Eth1/1/23   23    100000    Eth23-ASIC1  Ext
Ethernet25      768,769,770,771       Eth1/1/25   25    100000    Eth25-ASIC1  Ext
Ethernet26      772,773,774,775       Eth1/1/26   26    100000    Eth26-ASIC1  Ext
Ethernet28      776,777,778,779       Eth1/1/28   28    100000    Eth28-ASIC1  Ext
Ethernet29      780,781,782,783       Eth1/1/29   29    100000    Eth29-ASIC1  Ext
Ethernet31      512,513,514,515       Eth1/1/31   31    100000    Eth31-ASIC1  Ext
Ethernet32      516,517,518,519       Eth1/1/32   32    100000    Eth32-ASIC1  Ext
Ethernet34      520,521,522,523       Eth1/1/34   34    100000    Eth34-ASIC1  Ext
Ethernet35      524,525,526,527       Eth1/1/35   35    100000    Eth35-ASIC1  Ext
Ethernet37      256,257,258,259       Eth1/1/37   37    100000    Eth37-ASIC1  Ext
Ethernet38      260,261,262,263       Eth1/1/38   38    100000    Eth38-ASIC1  Ext
Ethernet40      268,269,270,271       Eth1/1/40   40    100000    Eth40-ASIC1  Ext
Ethernet41      264,265,266,267       Eth1/1/41   41    100000    Eth41-ASIC1  Ext
Ethernet43      12,13,14,15       Eth1/1/43   43    100000    Eth43-ASIC1  Ext
Ethernet44      8,9,10,11       Eth1/1/44   44    100000    Eth44-ASIC1  Ext
Ethernet46      4,5,6,7       Eth1/1/46   46    100000    Eth46-ASIC1  Ext
Ethernet47      0,1,2,3       Eth1/1/47   47    100000    Eth47-ASIC1  Ext
Ethernet2320      16,17       Eth2320-ASIC1   2320    100000   Eth2320-ASIC1   Int
Ethernet2452      2052,2053       Eth2452-ASIC1   2452    100000   Eth2452-ASIC1   Int
Ethernet2454      2054,2055       Eth2454-ASIC1   2454    100000   Eth2454-ASIC1   Int
Ethernet2456      2056,2057       Eth2456-ASIC1   2456    100000   Eth2456-ASIC1   Int
Ethernet2458      2058,2059       Eth2458-ASIC1   2458    100000   Eth2458-ASIC1   Int
Ethernet2460      2060,2061       Eth2460-ASIC1   2460    100000   Eth2460-ASIC1   Int
Ethernet2462      2062,2063       Eth2462-ASIC1   2462    100000   Eth2462-ASIC1   Int
Ethernet2464      2064,2065       Eth2464-ASIC1   2464    100000   Eth2464-ASIC1   Int
Ethernet2466      2304,2305       Eth2466-ASIC1   2466    100000   Eth2466-ASIC1   Int
Ethernet2468      2306,2307       Eth2468-ASIC1   2468    100000   Eth2468-ASIC1   Int
Ethernet2470      2308,2309       Eth2470-ASIC1   2470    100000   Eth2470-ASIC1   Int
Ethernet2472      2310,2311       Eth2472-ASIC1   2472    100000   Eth2472-ASIC1   Int
Ethernet2474      2312,2313       Eth2474-ASIC1   2474    100000   Eth2474-ASIC1   Int
Ethernet2476      2314,2315       Eth2476-ASIC1   2476    100000   Eth2476-ASIC1   Int
Ethernet2478      2316,2317       Eth2478-ASIC1   2478    100000   Eth2478-ASIC1   Int
Ethernet2480      2318,2319       Eth2480-ASIC1   2480    100000   Eth2480-ASIC1   Int
Ethernet2482      2320,2321       Eth2482-ASIC1   2482    100000   Eth2482-ASIC1   Int
Ethernet2484      2560,2561       Eth2484-ASIC1   2484    100000   Eth2484-ASIC1   Int
Ethernet2486      2562,2563       Eth2486-ASIC1   2486    100000   Eth2486-ASIC1   Int
Ethernet2488      2564,2565       Eth2488-ASIC1   2488    100000   Eth2488-ASIC1   Int
Ethernet2490      2566,2567       Eth2490-ASIC1   2490    100000   Eth2490-ASIC1   Int
Ethernet2492      2568,2569       Eth2492-ASIC1   2492    100000   Eth2492-ASIC1   Int
Ethernet2494      2570,2571       Eth2494-ASIC1   2494    100000   Eth2494-ASIC1   Int
Ethernet2496      2572,2573       Eth2496-ASIC1   2496    100000   Eth2496-ASIC1   Int
Ethernet2498      2574,2575       Eth2498-ASIC1   2498    100000   Eth2498-ASIC1   Int
Ethernet2500      2576,2577       Eth2500-ASIC1   2500    100000   Eth2500-ASIC1   Int
Ethernet2502      2816,2817       Eth2502-ASIC1   2502    100000   Eth2502-ASIC1   Int
Ethernet2504      2818,2819       Eth2504-ASIC1   2504    100000   Eth2504-ASIC1   Int
Ethernet2506      2820,2821       Eth2506-ASIC1   2506    100000   Eth2506-ASIC1   Int
Ethernet2508      2822,2823       Eth2508-ASIC1   2508    100000   Eth2508-ASIC1   Int
Ethernet2510      2824,2825       Eth2510-ASIC1   2510    100000   Eth2510-ASIC1   Int
Ethernet2512      2826,2827       Eth2512-ASIC1   2512    100000   Eth2512-ASIC1   Int
Ethernet2514      2828,2829       Eth2514-ASIC1   2514    100000   Eth2514-ASIC1   Int
Ethernet2516      2830,2831       Eth2516-ASIC1   2516    100000   Eth2516-ASIC1   Int
Ethernet2518      2832,2833       Eth2518-ASIC1   2518    100000   Eth2518-ASIC1   Int

Config db schema generated by minigraph can’t pass yang validation:

* there's no 'asic_port_name' and 'role' in sonic-port.yang.
* index range should be from 1 to 256.

It seems that we need to update sonic-config-engine or yang model.

#### Steps to reproduce the issue:
1. Run command 'sonic-cfggen -m tests/sample-chassis-packet-lc-graph.xml -p tests/sample-chassis-packet-lc-port-config.ini --print-data'.
2. Use sonic_yang to validate config db schema.

#### Describe the results you received:
Yang validation failed.

#### Describe the results you expected:
Yang validation should pass.

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

### PR DESCRIPTION
**Pull request by ganglyu**
Signed-off-by: Gang Lv ganglv@microsoft.com

#### Why I did it
Multi-asic platform add aisc_port_name and role to PORT table, and port_index range is changed.

#### How I did it
Update sonic-port.yang, add asic_port_name and role, and remove range limitation.

#### How to verify it
Run UT for sonic-yang-models.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog

Fix #9580

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-port.yang [hunk 1]
```diff
@@ -110,9 +110,19 @@ module sonic-port{
...
-						range 0..256;
+					type uint16;
+				}
+
+				leaf asic_port_name {
+					type string;
+					description "port name in asic and asic name";
+				}
+
+				leaf role {
```
Comment (dgsudharsan) [inline src/sonic-yang-models/yang-models/sonic-port.yang:121]: Should the role be valid only when asic_port_name is defined?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: src/sonic-yang-models/yang-models/sonic-port.yang (hunk 1 above)
Comment (ganglyu) [reply src/sonic-yang-models/yang-models/sonic-port.yang:121]: @SuvarnaMeenakshi would you please confirm? Role is valid only when asic_port_name is defined?

Comment (SuvarnaMeenakshi) [reply src/sonic-yang-models/yang-models/sonic-port.yang:121]: Yes, currently we only have role defined for asic and not for the device.
But as this field is not mandatory, defining it this way should be fine.
@arlakshm - please correct me if I am wrong.

File: src/sonic-yang-models/yang-models/sonic-port.yang [hunk 2]
```diff
@@ -110,9 +110,19 @@ module sonic-port{
...
 
 				leaf index {
-					type uint16 {
-						range 0..256;
+					type uint16;
+				}
+
+				leaf asic_port_name {
+					type string;
+					description "port name in asic and asic name";
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-port.yang:118]: This description is not clear, please provide an example.

**Review by dgsudharsan — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-port.yang (hunk 1 above)
Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-port.yang:121]: @dgsudharsan, @venkatmahalingam. The `role` is not related to `asic_port_name`. 
The `role` indicates if the port is an internal port or external(frontpanel) facing port. if a `role` is not specified then the port is assumed to be external facing
The 'asic_port_name` indicates the port on the packet processer. It is used to map Sonic port name to the port on the asic

IMO, these two are attributes not related and we should not have a condition that the role to  asic port_name and vice versa.

**Review by venkatmahalingam — DISMISSED** [review]

**Comment by ganglyu** [comment]
@SuvarnaMeenakshi @arlakshm would you please review this PR?

File: src/sonic-yang-models/tests/yang_model_tests/tests/port.json [hunk 3]
```diff
@@ -73,5 +73,13 @@
      "PORT_INVALID_MUX_CABLE_TEST": {
          "desc": "PORT_INVALID_MUX_CABLE_TEST non-boolean values, expect fail",
          "eStrKey": "InvalidValue"
+     },
+     "PORT_VALID_MULTIASIC_TEST": {
+         "desc": "PORT_VALID_MULTIASIC_TEST no failure."
+     },
+     "PORT_INVALID_MULTIASIC_TEST": {
```
Comment (SuvarnaMeenakshi) [inline src/sonic-yang-models/tests/yang_model_tests/tests/port.json:80]: Is there a list of values that are considered to be invalid?

File: src/sonic-yang-models/yang-models/sonic-port.yang [hunk 4]
```diff
@@ -110,9 +110,19 @@ module sonic-port{
...
+				}
+
+				leaf asic_port_name {
+					type string;
+					description "port name in asic and asic name, e.g Eth0-ASIC1";
+				}
+
+				leaf role {
+					type string {
+						pattern "Ext|Int";
```
Comment (SuvarnaMeenakshi) [inline src/sonic-yang-models/yang-models/sonic-port.yang:123]: @arlakshm should the pattern list include "Inb" or "Rec" ?

File: src/sonic-yang-models/tests/yang_model_tests/tests/port.json (hunk 3 above)
Comment (SuvarnaMeenakshi) [reply src/sonic-yang-models/tests/yang_model_tests/tests/port.json:80]: Any string other than the ones added in "pattern" for role is considered to be invalid?

File: src/sonic-yang-models/yang-models/sonic-port.yang (hunk 4 above)
Comment (SuvarnaMeenakshi) [reply src/sonic-yang-models/yang-models/sonic-port.yang:123]: Also, can there be a default value "Ext" for this?

File: src/sonic-yang-models/tests/yang_model_tests/tests/port.json (hunk 3 above)
Comment (ganglyu) [reply src/sonic-yang-models/tests/yang_model_tests/tests/port.json:80]: Yes, string must match the pattern

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**